tasks/internal/update-fbc-catalog-task/tests/mocks.sh

```diff
@@ -65,23 +63,63 @@ function mock_build_progress() {
+      if [[ "$extracted_from_index" =~ registry-proxy.engineering.redhat.com/rh-osbs/iib-pub ]]; then
+        is_registry_proxy_test=true
+      fi
+    fi
     build="${buildSeed}"
-    echo "DEBUG: Checking previous builds, taskrun name: $(context.taskRun.name)" >&2
-    case "$(context.taskRun.name)" in
+    taskrun_name="$(context.taskRun.name)"
+    
+    # Check if this is the auth-failure test by from_index value FIRST (more reliable)
+    # If from_index contains registry-proxy, treat it as auth-failure regardless of taskrun name
+    # For auth-failure test: return an OLD build (>24 hours) that should be rejected
+    if [[ "$is_registry_proxy_test" = "true" ]] || \
+       ([[ -n "$extracted_from_index" ]] && [[ "$extracted_from_index" =~ registry-proxy.engineering.redhat.com/rh-osbs/iib-pub ]]); then
+      # Set updated timestamp to 25 hours ago (old build that should be rejected)
+      old_timestamp=$(date -u -d '25 hours ago' --iso-8601=seconds 2>/dev/null || date -u -v-25H --iso-8601=seconds 2>/dev/null || echo "$(date -u --iso-8601=seconds)")
+      jq_expr='.items[0] |= (.fbc_fragments = ["registry.io/image0@sha256:0000"] | .from_index = "registry-proxy.engineering.redhat.com/rh-osbs/iib-pub:v4.12" | .from_index_resolved = "registry-proxy.engineering.redhat.com/rh-osbs/iib-pub@sha256:3d6fe02b28ab876d60af3c3df5100a6fe4c99b084651af547659173d680c6f4d" | .index_image = "registry-proxy.engineering.redhat.com/rh-osbs/iib-pub:v4.12" | .index_image_resolved = "registry-proxy.engineering.redhat.com/rh-osbs/iib-pub@sha256:10b1d5b1d053d8c3a2263201baa983c760e6b61f3a50e3cde244fdaf68a4aed9" | .updated = "'"${old_timestamp}"'" | .state = "complete" | .state_reason = "The FBC fragment was successfully added in the index image")'
+      build=$(jq -rc "$jq_expr" <<< "${build}")
+    else
+      # Fall back to taskrun name matching
+      case "${taskrun_name}" in
+        *auth-failure*)
+          # For auth-failure test, return an OLD build (>24 hours) that should be rejected
+          # Check this BEFORE *complete* to avoid pattern collision
+          old_timestamp=$(date -u -d '25 hours ago' --iso-8601=seconds 2>/dev/null || date -u -v-25H --iso-8601=seconds 2>/dev/null || echo "$(date -u --iso-8601=seconds)")
+          jq_expr='.items[0] |= (.fbc_fragments = ["registry.io/image0@sha256:0000"] | .from_index = "registry-proxy.engineering.redhat.com/rh-osbs/iib-pub:v4.12" | .from_index_resolved = "registry-proxy.engineering.redhat.com/rh-osbs/iib-pub@sha256:3d6fe02b28ab876d60af3c3df5100a6fe4c99b084651af547659173d680c6f4d" | .index_image = "registry-proxy.engineering.redhat.com/rh-osbs/iib-pub:v4.12" | .index_image_resolved = "registry-proxy.engineering.redhat.com/rh-osbs/iib-pub@sha256:10b1d5b1d053d8c3a2263201baa983c760e6b61f3a50e3cde244fdaf68a4aed9" | .updated = "'"${old_timestamp}"'" | .state = "complete" | .state_reason = "The FBC fragment was successfully added in the index image")'
```

Comment:
maybe should rewrite in some way to reuse this string as it appears in the block above as well.